master/buildbot/changes/gitpoller.py

```diff
@@ -481,37 +484,86 @@ def _process_changes(self, newRev, branch):
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth.ssh_host_key
+
+    @property
+    def sshKnownHosts(self):
+        ssh_auth_warndeprecate(
+            "self.sshKnownHosts",
+            "self._git_ssh_auth.ssh_known_hosts",
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth.ssh_known_hosts
+
     def _isSshPrivateKeyNeededForCommand(self, command):
-        commandsThatNeedKey = [
-            'fetch',
-            'ls-remote',
-        ]
-        if self.sshPrivateKey is not None and command in commandsThatNeedKey:
-            return True
-        return False
+        ssh_auth_warndeprecate(
+            "self._isSshPrivateKeyNeededForCommand",
+            "self._git_ssh_auth.is_ssh_private_key_needed_for_git_command",
+        )
+        return (
+            self._git_ssh_auth is not None and
+            self._git_ssh_auth.is_ssh_private_key_needed_for_git_command(command)
+        )
 
     def _downloadSshPrivateKey(self, keyPath):
+        ssh_auth_warndeprecate(
+            "self._downloadSshPrivateKey",
+            "self._git_ssh_auth.download_ssh_private_key_if_needed",
+        )
         # We change the permissions of the key file to be user-readable only so
         # that ssh does not complain. This is not used for security because the
         # parent directory will have proper permissions.
         writeLocalFile(keyPath, ensureSshKeyNewline(self.sshPrivateKey), mode=stat.S_IRUSR)
 
     def _downloadSshKnownHosts(self, path):
+        ssh_auth_warndeprecate(
+            "self._downloadSshKnownHosts",
+            "self._git_ssh_auth.download_ssh_private_key_if_needed",
+        )
         if self.sshKnownHosts is not None:
             contents = self.sshKnownHosts
         else:
             contents = getSshKnownHostsContents(self.sshHostKey)
         writeLocalFile(path, contents)
 
     def _getSshPrivateKeyPath(self, ssh_data_path):
-        return os.path.join(ssh_data_path, 'ssh-key')
+        ssh_auth_warndeprecate(
+            "self._getSshPrivateKeyPath",
+            "self._git_ssh_auth._get_ssh_private_key_path",
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth._get_ssh_private_key_path(ssh_data_path)
 
     def _getSshKnownHostsPath(self, ssh_data_path):
-        return os.path.join(ssh_data_path, 'ssh-known-hosts')
+        ssh_auth_warndeprecate(
+            "self._getSshKnownHostsPath",
+            "self._git_ssh_auth._get_ssh_host_key_path",
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth._get_ssh_host_key_path(ssh_data_path)
 
     @defer.inlineCallbacks
     def _dovccmd(self, command, args, path=None):
-        if self._isSshPrivateKeyNeededForCommand(command):
+        if (
+            self._git_ssh_auth is not None and
+            self._git_ssh_auth.is_ssh_private_key_needed_for_git_command(command)
```

Comment:
There is some checks as this one that are not great in themselves.
I intend to hide them in a `git_auth_handler` which will allow to add auth by user:password/token mentioned in #7384 in a fairly transparent manner later on.